### DISCUSSION TITLE
Help to run blockscout with ganache and docker, for local testing?

### DISCUSSION THREAD

**Discussion by cryptotester** [blockscout/blockscout#5745] [Support] [answered]
Hi,

I'm using Fedora Linux 36 and using ganache-cli v 7.3.1

I'm running ganache-cli with some options to persist the data, as you can see I use 5777 as network id, and I save the ganache db in a folder called ganache_test:

```
ganache-cli --db ganache_test -i 5777 --accounts 20 --deterministic --mnemonic="my mnemonic"
```

Now I try to run blockscout with docker, following your docu here: https://docs.blockscout.com/for-developers/information-and-settings/docker-integration-local-use-only

Instead of using port 2000 I use 8545 (default ganache port). I cloned the blockscout repo and I `cd docker`, then run

```
COIN=ETH \
ETHEREUM_JSONRPC_VARIANT=ganache \
ETHEREUM_JSONRPC_HTTP_URL=http://host.docker.internal:8545 \
ETHEREUM_JSONRPC_WS_URL=ws://host.docker.internal:8545 \
make start
```

Blockscout starts and  I can browse it at http://localhost:4000/

I use truffle to develop and test smart contracts. In my `truffle-config.js` file I did configure a network in able to use my persistent ganache_test instance, here the config:

```
ganache: {
  host: "127.0.0.1",
  port: 8545,
  network_id: "5777",
},
```

In truffle, I run: `truffle migrate --network ganache --reset`
This redeploys my smart contracts in my ganache network I started above. I can see in the logs that new blocks are created etc.

I refresh http://localhost:4000/ and I see nothing, no transactions, no blocks
Expected: I expect to see new blocks, transactions etc. I mean this is the whole point of blockscout...

I searched and couldn't find any tutorial or guide to run blockscout with ganache. I tried to make my post like a tutorial, it lacks of course some info in order to make it work.

I'd be very pleased if someone helps me to add the missing pieces to make my local setup work.

Also, how to reset the posgres db? I mean to start from scratch I'd like to reset the db, maybe with my various attempts I did mess up something, e.g. before I tried a ganache with different network id, different port etc.

Thank you

**Comment by vbaranov** [comment, marked as answer]
Resolved in frame of https://github.com/blockscout/blockscout/issues/5951
